app/controllers/users_controller.rb

```diff
@@ -0,0 +1,51 @@
+class UsersController < ApplicationController
+  enable :sessions
+  set :session_secret, 'fwitter'
+  set :public_folder, 'public'
+  set :views, 'app/views'
+
+# Displayse the "Sign Up" form and creates a new user
+  get '/signup' do
+    redirect '/tweets' if logged_in?
+    erb :'users/create'
+  end
+
+  post '/signup' do
+    redirect '/signup' if params['username'] == "" || params['email'] == "" || params['password'] == ""
```

Comment:
The lesson you linked to doesn't mention that method of validating.
By coincidence, I was just part of session with Luke where he mentioned it so I googled it and learned something new :)